cumulus/pallets/parachain-system/src/lib.rs

```diff
@@ -914,6 +923,12 @@ pub mod pallet {
 	#[pallet::storage]
 	pub type CustomValidationHeadData<T: Config> = StorageValue<_, Vec<u8>, OptionQuery>;
 
+	/// The claim queue offset.
+	/// This value is expected to be mostly a constant. It's used by collators to determine when to
+	/// build a collation.
+	#[pallet::storage]
+	pub type ClaimQueueOffset<T: Config> = StorageValue<_, u8>;
```

Comment:
at the moment of the relay chain parent. When the block gets backed on chain, the paraid must be on the first position in the claim queue. so it basically enables a parachain to build ahead of time, seeing that in X blocks it will be on the top position in the claim queue

Reply:
>  so it basically enables a parachain to build ahead of time, seeing that in X blocks it will be on the top position in the claim queue

We could do this even before RFC 103, we could build on a core for which we see a claim in the future, but collators did not take advantage of it.

The claim queue offset is relative to the relay parent. And the core selector is an index to the cores assigned to the parachain at the specified claim queue offset. It provides a deterministic way for parachain runtimes to commit to a core index on the relay chain without actually looking at the claim queue.